components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -267,18 +175,25 @@ GURL AIChatTabHelper::GetPageURL() const {
 void AIChatTabHelper::GetPageContent(GetPageContentCallback callback,
                                      std::string_view invalidation_token) {
   bool is_pdf = IsPdf(web_contents());
-  if (is_pdf && !is_pdf_a11y_info_loaded_) {
-    SetPendingGetContentCallback(std::move(callback));
-    // Manually check when pdf extraction requested so we don't always rely on
-    // a11y events to prevent stale callback. It can happens during background
-    // pdf tab loading or bug in upstream kPdfOCR that an empty page in pdf will
-    // cause PdfOcrHelper::AreAllPagesOcred() return false becasue it won't get
-    // Ocred but `remaining_page_count_` was set to total pdf page count. Hence
-    // status IDS_PDF_OCR_COMPLETED or IDS_PDF_OCR_NO_RESULT won't be set.
-    CheckPDFA11yTree();
-    return;
+  bool pdf_helper_not_available = false;
+  if (is_pdf) {
+#if BUILDFLAG(ENABLE_PDF)
+    auto* pdf_helper =
+        pdf::PDFDocumentHelper::MaybeGetForWebContents(web_contents());
+    if (pdf_helper) {
+      pdf_helper->RegisterForDocumentLoadComplete(base::BindOnce(
+          &AIChatTabHelper::OnPDFDocumentLoadComplete,
+          weak_ptr_factory_.GetWeakPtr(),
+          base::BindOnce(&AIChatTabHelper::OnFetchPageContentComplete,
+                         weak_ptr_factory_.GetWeakPtr(), std::move(callback))));
+      return;
+    } else {
+      pdf_helper_not_available = true;
+    }
+#endif
   }
-  if (kPrintPreviewRetrievalHosts.contains(GetPageURL().host_piece())) {
+  if (kPrintPreviewRetrievalHosts.contains(GetPageURL().host_piece()) ||
+      pdf_helper_not_available) {
```

Comment:
On Android without pdf_helper, it seems we won't fallback to this because pdf_helper_not_available is not set to true. We should cover the fallback case for this in unit test, guarded by #if !BUILDFLAG(ENABLE_PDF).

Reply:
I suppose we want to fallback to this when enable_pdf is false.